### PR TITLE
create ui updates for nav bar and map and maintain map state

### PR DESCRIPTION
**Pull request by alyssa-blair**
This pr updates the map UI to resemble the mockups more closely. It also fixes the issue with the app crashing when you type only a space into the search bar. The MapViewModel class allows us to save the state of the map when switching between pages.

Pasted Graphic

### CODE REVIEW

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt [hunk 1]
```diff
@@ -9,40 +9,41 @@ import androidx.compose.foundation.layout.fillMaxWidth
...
+import androidx.compose.runtime.*
 
 /**
  * A composable that displays a map with a search box and suggestions.
  *
  * @param spots The spots to display on the map.
  */
 @Composable
-fun MapView(spots: List<Spot>) {
+fun MapView(spots: List<Spot>, mapViewModel: MapViewModel) {
```
Comment (wrighted) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt:35]: This should probably be part of the ViewModel

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt [hunk 2]
```diff
@@ -9,40 +9,41 @@ import androidx.compose.foundation.layout.fillMaxWidth
...
     val scope = rememberCoroutineScope()
     val suggestionGenerator = remember { SearchBoxFetcher() }
+    val focusManager = LocalFocusManager.current
 
-    var query by remember { mutableStateOf("") }
-    var suggestions by remember { mutableStateOf(emptyList<Suggestion>()) }
-    var currentLocation by remember { mutableStateOf(Coordinate.DEFAULT) }
+    val query by mapViewModel.query.collectAsState("")
+    val suggestions by mapViewModel.suggestions.collectAsState(emptyList())
+    val currentLocation by mapViewModel.currentLocation.collectAsState(Coordinate.DEFAULT)
```
Comment (wrighted) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt:42]: Should these not still be in `remember { }` blocks?

**Review by wrighted — APPROVED** [review]
Minor nits but seems good.

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt (hunk 2 above)
Comment (alyssa-blair) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt:42]: probably will have to look back to this later, remember blocks don't seem to work well with collectAsState.

**Review by wrighted — APPROVED** [review]
Need to look at not reloading the map when switching screens, and moving the `spots` dependency to be _constructed_ with the ViewModel.

Both can be done at a later time.

**Pull request merged**